### DISCUSSION TITLE
Unable to parse column text

### DISCUSSION THREAD

**Discussion by kakuibeyond** [jsvine/pdfplumber#747] [Ask for help with specific PDFs]
Hi all:
When I use pdfplumber to parse my pdf document, the text result I get is wrong, how should I modify the code to get the desired columnar result.
- mypdf document as shown below
![image](https://user-images.githubusercontent.com/53488265/197500426-ef0b54f8-79be-44c2-9b9c-e8dd10285047.png)
[PDF test file.pdf](https://github.com/jsvine/pdfplumber/files/9850945/PDF.test.file.pdf)

- my parse code
```python
  with pdfplumber.open(pdf_path) as pdf:
      all_pages = pdf.pages
      raw_text = all_pages[0].extract_text()
      print(raw_text)
```
- output
```json
PDFtestﬁle
快速引导
QUICKGUIDE
安装电池 ⼿柄开机
拇指按压电池 短按⼿柄HOME键盘
拨出绝缘纸
```
- my desired result
```json
PDFtestﬁle
快速引导
QUICKGUIDE
安装电池
拇指按压电池拨出绝缘纸
⼿柄开机
短按⼿柄HOME键盘
```

**Comment by jsvine** [comment]
Hi @kakuibeyond, and thanks for your interest in this library. I can suggest two options:

- Use `page.crop((0, 0, page.width/2, page.height)).extract_text()` to get the data from the left-hand side of the page, and then `page.crop((page.width/2, 0, page.width, page.height)).extract_text()` again on the right-hand side. 

- Use `page.extract_text(layout=True)` plus regular expressions.
